SECURITY.md

```diff
@@ -1,6 +1,63 @@
+ - [**Reporting a Vulnerability**](#reporting-a-vulnerability)
+
+## Using Pytorch Securely
+**Consider Pythor models as programs**. Machine Learning models are mathematical vectors and weights in their essence, but as they grow in complexity they can require privileges or be partially composed by programmatic code. That said, running untrusted models is equivelent to running untrusted code.
+
+### Untrusted models
+Be careful when running untrusted models. This classification includes models created by unknown developers or utilizing data obtained from unknown sources.
+
+**Prefer to execute untrusted models within a secure, isolated environment such as a sandbox** (e.g., containers, virtual machines). This helps protect your system from potentially malicious code. You can find further details and instructions in [this page](https://developers.google.com/code-sandboxing).
+
+**Be mindful of risky model formats**. Give preference to run models with `.safetensors` format, and if you need to run load models on the [picle](https://docs.python.org/3/library/pickle.html) format, make sure to use the `weights_only` option to `torch.load`.
+
+Important Note: The trustworthiness of a model is not binary. You must always determine the proper level of caution depending on the specific model and how it matches your use case and risk tolerance.
+
+### Untrusted inputs during training and prediction
+
+Some models accept various input formats (text, images, audio, etc.) and pre-process them by modifying and/or converting the input. Most of this work is done by a variaty of libraries that provide different levels of security. Based on the security history of these libraries, it's considered safe to work with untrusted inputs for PNG, BMP, GIF, WAV, RAW, RAW_PADDED, CSV and PROTO formats. If you process untrusted data in any other format, we you should employ the following recommendations:
+
+* Sandboxing: Isolate the model process. You can follow the same instructions as given on the section [Untrusted models](#untrusted-models).
+* Updates: Keep your model and libraries updated with the latest security patches.
+
+### Data privacy
+
+**Take special security measures if your model if you train models with sensitive data**. Prioritize [sandboxing](https://developers.google.com/code-sandboxing) your models and:
+- Do not feed sensitive data to untrusted model (even if runs in a sandboxed environment)
+- If you consider publishing a model that was partially trained with sensitive data, be aware that data can potentially be recovered from the trained weights (especially if model overfits).
+
+### Untrusted environments or networks
+
+If you can't run your models in a secure and isolated environment or if it must be exposed to an untrusted network, make sure to take the following security precautions:
+* Confirm the hash of any downloaded artifact (e.g. pre-trained model weights) matches a known-good value
```

Comment:
This feels oddly specific? If we want to cover how to handle unsecure env/network, I guess this list should be much longer. I'm not sure we can even be exhaustive here given the breadth of users we have so I think that pointing out the risk clearly is enough without providing solution to specific problems that might not apply.

Reply:
Understood, it makes sense to me too. I'm now considering to remove the `Untrusted environment or networks` and `Multi-Tenant environments` sections, does that sound good to you?

Reply:
I am happy with either:
- Having a single link to a much more in-depth third party description (cloud provider analysis, research paper, ?)
- Remove it altogether if we don't have good material to link to.

Reply:
I'd rather remove it altogether, because now I realize those sections are pretty generalists and we should keep this security guidance specific for pytorch usage. Done, PTAL